onchain/cairo/afk/src/bip340.cairo

```diff
@@ -260,14 +268,28 @@ fn encodeSocialRequest<
 }
 
 /// Generates a key pair (private key, public key) for Schnorr signatures
-fn generate_keypair() -> (u256, Secp256k1Point) {
+fn generate_keypair() -> (core::felt252, core::starknet::secp256k1::Secp256k1Point) {
+    // vrf address
+    let vrf_provider = IVrfProviderDispatcher {
+        contract_address: starknet::contract_address_const::<0x123>()
```

Comment:
Please, change the address with a params inputs in the function, for the VRF we sent on the call